librad/src/net/protocol.rs

```diff
@@ -189,11 +189,14 @@ where
     /// Note that responses will also cause [`gossip::LocalStorage::put`] to be
     /// invoked, i.e. the local storage will be converged towards the
     /// requested state.
-    pub async fn query(&self, want: A) -> impl futures::Stream<Item = gossip::Has<A>> {
-        self.gossip.query(want).await;
-        self.gossip.subscribe().await.filter_map(|evt| async move {
+    pub async fn query(&self, want: A) -> impl futures::Stream<Item = gossip::Has<A>>
+    where
+        A: PartialEq + Clone,
+    {
+        self.gossip.query(want.clone()).await;
+        self.gossip.subscribe().await.zip(futures::stream::repeat(want)).filter_map(|(evt, wanted)| async move {
```

Comment:
Curious why you replicate `want` instead of checking the `has` with a captured `want`? Is it because it's not `Sync + Send`?

Reply:
See rant in IRC 😬